### ISSUE TITLE
Trying to eager load associated objects using 'include' for a belongsToMany relationship

### ISSUE THREAD

**Issue by ashishsanjayrao** [sequelize/sequelize#11443]
## Issue Description
I have a model called Experience, which has a belongsToMany association with the model Category through ExperienceCategory table. For an experience, I'm trying to fetch the Experience details with the Categories it belongs to. Some I'm using include: `[{model: model.Category}]` to achieve this. Now, the association is working and it is fetching the categories correctly, but the problem is that it's giving multiple objects of the Experience model, one for each Category. Below are some excerpts from my code:

### What are you doing?
```
model.Experience.findAll({
  include: [
    {
      model: model.Category
    }
  ]
}).then(experiences => {
  res.json({experiences: experiences});
})
```
This is how I have defined my associations:<br>
experience.js:
```
Experience.belongsToMany(models.Category, {
  through: 'ExperienceCategory',
  foreignKey: 'experience_id',
});
```
category.js
```
Category.belongsToMany(models.Experience, {
  through: 'ExperienceCategory',
  foreignKey: 'category_id'
});
```
### What is actually happening?
```
{
  "experiences": [
    {
      "id": 23,
      "title": "Some title",
      "notes": "These are notes",
      "base_amount": 2000,
      "is_disabled": false,
      "is_deleted": false,
      "createdAt": "2019-05-27T12:24:06.736Z",
      "updatedAt": "2019-07-23T11:20:23.695Z",
      "Categories.id": 4,
      "Categories.name": "Trekking",
      "Categories.is_disabled": false,
      "Categories.is_deleted": false,
      "Categories.createdAt": "2019-05-14T11:07:40.287Z",
      "Categories.updatedAt": "2019-05-14T11:07:40.287Z",
      "Categories.ExperienceCategory.experience_id": 23,
      "Categories.ExperienceCategory.category_id": 4,
      "Categories.ExperienceCategory.createdAt": "2019-05-27T12:24:06.806Z",
      "Categories.ExperienceCategory.updatedAt": "2019-05-27T12:24:06.806Z"
    },
    {
      "id": 23,
      "title": "Some title",
      "notes": "These are notes",
      "base_amount": 2000,
      "is_disabled": false,
      "is_deleted": false,
      "createdAt": "2019-05-27T12:24:06.736Z",
      "updatedAt": "2019-07-23T11:20:23.695Z",
      "Categories.id": 5,
      "Categories.name": "Adventure",
      "Categories.is_disabled": false,
      "Categories.is_deleted": false,
      "Categories.createdAt": "2019-05-14T11:07:40.287Z",
      "Categories.updatedAt": "2019-05-14T11:07:40.287Z",
      "Categories.ExperienceCategory.experience_id": 23,
      "Categories.ExperienceCategory.category_id": 5,
      "Categories.ExperienceCategory.createdAt": "2019-05-27T12:24:06.806Z",
      "Categories.ExperienceCategory.updatedAt": "2019-05-27T12:24:06.806Z"
    }
  ]
}
```

### Environment

- Sequelize version: 5.8.6
- Node.js version: 10.15.3

### How does this problem relate to dialects?

- [x] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s): 
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by papb** [comment]
Hello, looks like you found a solution already, can you post a short explanation for it, in case others have the same problem in the future? Thanks

**Comment by ashishsanjayrao** [comment]
Sure. It was just an overlook from my side. I had set `raw: true` in the query, which gave me multiple instances of Experience. After I removed it, I got the Categories in the form of an array for each Experience. 

When I posted the bug here, I typed out the query instead of copying and pasting it from my code and never imagined that that could be the problem.

Hope this helps somebody in the future, but I hope no one makes a silly mistake like this one.
